msbuild/Messaging/Xamarin.Messaging.Build/Program.cs

```diff
@@ -10,7 +12,14 @@ static async Task Main (string [] args)
 			var agent = new BuildAgent (topicGenerator, arguments.Version, arguments.VersionInfo);
 			var runner = new AgentConsoleRunner<BuildAgent> (agent, arguments);
 
-			await runner.RunAsync ().ConfigureAwait (continueOnCapturedContext: false);
+			//Hack to support legacy paths from Windows (likely Dev17 versions)
+			if (MessagingContext.BasePath.Contains ("Xamarin")) {
+				var xamarinPath = MessagingContext.BasePath.Substring (0, MessagingContext.BasePath.IndexOf ("Xamarin") + "Xamarin".Length);
```

Comment:
A minor perf improvement:

```suggestion
			var idx = MessagingContext.BasePath.IndexOf ("Xamarin", StringComparison.Ordinal);
			if (idx >= 0) {
				var xamarinPath = MessagingContext.BasePath.Substring (0, idx + "Xamarin".Length);
```

Reply:
Fixed in the most recent commit